sdks/java/io/kafka/src/main/java/org/apache/beam/sdk/io/kafka/KafkaIO.java

```diff
@@ -1047,7 +1047,9 @@ public void setMaxReadTime(Long maxReadTime) {
       // "beam_fn_api_use_deprecated_read" is not enabled.
       if (!ExperimentalOptions.hasExperiment(input.getPipeline().getOptions(), "beam_fn_api")
           || ExperimentalOptions.hasExperiment(
-              input.getPipeline().getOptions(), "beam_fn_api_use_deprecated_read")) {
+              input.getPipeline().getOptions(), "beam_fn_api_use_deprecated_read")
+          || !ExperimentalOptions.hasExperiment(
+              input.getPipeline().getOptions(), "use_sdf_kafka_read")) {
```

Comment:
You'll need to add this experiment to the impacted tests since some tests expect an SDF.

Reply:
I couldn't  find a way to pass experiments into ExpansionService: https://github.com/apache/beam/blob/master/sdks/java/expansion-service/src/main/java/org/apache/beam/sdk/expansion/service/ExpansionService.java#L384-L393 so I just leave the test to verity non-sdf read.